Unpin libcxx in Conda recipes

**Describe the outcome that is desired.**
Unpin libcxx in the Conda recipes (see https://github.com/mantidproject/mantid/pull/37307)

**Describe any solutions you are considering**
The pin was needed to prevent a build failure when compiling Framework/Kernel/src/TimeROI.cpp. It might be possible to fix the issue here. Most likely [#37310](https://github.com/mantidproject/mantid/pull/37310) is the only fix needed for this.

**Additional context**
`Framework/Kernel/CMakeFiles/Kernel.dir/src/TimeROI.cpp.o -MF Framework/Kernel/CMakeFiles/Kernel.dir/src/TimeROI.cpp.o.d -o Framework/Kernel/CMakeFiles/Kernel.dir/src/TimeROI.cpp.o -c /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/work/Framework/Kernel/src/TimeROI.cpp
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/work/Framework/Kernel/src/TimeROI.cpp:8:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/iostream:43:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/ios:221:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/__ios/fpos.h:14:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/iosfwd:106:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/__std_mbstate_t.h:14:
00:02:51  In file included from /Users/mantidbuilder/Jenkins/workspace/main_nightly_deployment/conda-bld/mantid_1715122430218/_build_env/bin/../include/c++/v1/__mbstate_t.h:45:
00:02:51  /opt/MacOSX10.10.sdk/usr/include/sys/_types/_mbstate_t.h:31:9: error: unknown type name '__darwin_mbstate_t'
00:02:51  typedef __darwin_mbstate_t mbstate_t;
00:02:51          ^
00:02:51  1 error generated.`

--- a/Framework/DataHandling/inc/MantidDataHandling/PDLoadCharacterizations.h
+++ b/Framework/DataHandling/inc/MantidDataHandling/PDLoadCharacterizations.h
@@ -6,11 +6,11 @@
 // SPDX - License - Identifier: GPL - 3.0 +
 #pragma once
 
-#include <iosfwd>
-
 #include "MantidAPI/Algorithm.h"
 #include "MantidAPI/ITableWorkspace_fwd.h"
 #include "MantidDataHandling/DllConfig.h"
+
+#include <iosfwd>
 
 namespace Mantid {
 namespace DataHandling {

--- a/Framework/DataHandling/src/LoadNexusLogs.cpp
+++ b/Framework/DataHandling/src/LoadNexusLogs.cpp
@@ -8,10 +8,9 @@
 #include "MantidAPI/FileProperty.h"
 #include "MantidAPI/LogManager.h"
 #include "MantidAPI/Run.h"
+#include "MantidDataHandling/LoadTOFRawNexus.h"
 #include "MantidKernel/ArrayProperty.h"
 #include "MantidKernel/TimeSeriesProperty.h"
-#include <locale>
-#include <nexus/NeXusException.hpp>
 
 #include <Poco/DateTimeFormat.h>
 #include <Poco/DateTimeFormatter.h>
@@ -19,10 +18,11 @@
 #include <Poco/Glob.h>
 #include <Poco/Path.h>
 
-#include "MantidDataHandling/LoadTOFRawNexus.h"
 #include <boost/scoped_array.hpp>
+#include <nexus/NeXusException.hpp>
 
 #include <algorithm>
+#include <locale>
 
 namespace Mantid::DataHandling {
 // Register the algorithm into the algorithm factory

--- a/conda/recipes/conda_build_config.yaml
+++ b/conda/recipes/conda_build_config.yaml
@@ -40,9 +40,6 @@ jemalloc:
 numpy:
   - 1.24.*
 
-libcxx:
-  - '<17'
-
 matplotlib:
   - 3.7.*
 

--- a/conda/recipes/mantid/meta.yaml
+++ b/conda/recipes/mantid/meta.yaml
@@ -30,7 +30,6 @@ requirements:
     - {{ cdt('libxxf86vm') }}                # [linux]
     - {{ cdt('libx11-devel') }}              # [linux]
     - {{ cdt('xorg-x11-proto-devel') }}      # [linux]
-    - libcxx  {{ libcxx }}                   # [osx]
 
   host:
     - eigen

--- a/conda/recipes/mantidqt/meta.yaml
+++ b/conda/recipes/mantidqt/meta.yaml
@@ -33,7 +33,6 @@ requirements:
     - {{ cdt('libxxf86vm') }}  # [linux]
     - {{ cdt('libx11-devel') }}  # [linux]
     - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
-    - libcxx  {{ libcxx }}                   # [osx]
 
   host:
     - libboost-devel {{ libboost_devel }}

--- a/mantid-developer-osx.yml
+++ b/mantid-developer-osx.yml
@@ -54,7 +54,7 @@ dependencies:
   - black  # may be out of sync with pre-commit
   - cppcheck==2.14.2
   - pre-commit>=2.12.0
-  - libcxx<17
+  - libcxx
 
   # Use conda version of clang to get matching CXXFLAGS.
   # Pinned to same version as conda forge: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml

--- a/mantid-developer-osx.yml
+++ b/mantid-developer-osx.yml
@@ -54,7 +54,7 @@ dependencies:
   - black  # may be out of sync with pre-commit
   - cppcheck==2.14.2
   - pre-commit>=2.12.0
-  - libcxx
+  - libcxx<17
 
   # Use conda version of clang to get matching CXXFLAGS.
   # Pinned to same version as conda forge: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml

